test/unit/pipeline/FetchUnitTest.cc

```diff
@@ -99,12 +113,328 @@ TEST_F(PipelineFetchUnitTest, FetchUnaligned) {
   fetchUnit.requestFromPC();
 
   // Tick again, expecting that decoding will now resume
-  MemoryReadResult nextBlockValue = {{16, 16}, 0, 1};
+  MemoryReadResult nextBlockValue = {{16, blockSize}, 0, 1};
   span<MemoryReadResult> nextBlock = {&nextBlockValue, 1};
-  EXPECT_CALL(memory, getCompletedReads()).WillOnce(Return(nextBlock));
-  EXPECT_CALL(isa, predecode(_, _, _, _))
-      .WillOnce(DoAll(SetArgReferee<3>(macroOp), Return(4)));
+  ON_CALL(memory, getCompletedReads()).WillByDefault(Return(nextBlock));
+  ON_CALL(isa, predecode(_, _, _, _))
+      .WillByDefault(DoAll(SetArgReferee<3>(mOp), Return(4)));
+  EXPECT_CALL(memory, getCompletedReads()).Times(1);
+  EXPECT_CALL(memory, clearCompletedReads()).Times(4);
+  EXPECT_CALL(isa, getMaxInstructionSize()).Times(8);
+  EXPECT_CALL(isa, predecode(_, _, _, _)).Times(4);
+
+  // Tick 4 times to process all 16 bytes of fetched data
+  for (int i = 0; i < 4; i++) {
+    fetchUnit.tick();
+  }
+  // Tick a 5th time to ensure all buffered bytes have been used
+  EXPECT_CALL(memory, getCompletedReads()).Times(1);
+  EXPECT_CALL(isa, getMaxInstructionSize()).Times(1);
+  EXPECT_CALL(isa, predecode(_, _, _, _)).Times(0);
+  fetchUnit.tick();
+}
+
+// Tests that a properly aligned PC (to the fetch block boundary) is correctly
+// fetched
```

Comment:
Should there be a test for a misaligned pc?

Reply:
There is - see `FetchUnaligned` above this test